httpbakery/error.go

```diff
@@ -91,14 +107,30 @@ func (e *Error) ErrorInfo() *ErrorInfo {
 // they encounter an error with a *bakery.Error cause.
 func ErrorToResponse(err error) (int, interface{}) {
 	errorBody := errorResponseBody(err)
+	var body interface{} = errorBody
 	status := http.StatusInternalServerError
 	switch errorBody.Code {
 	case ErrBadRequest:
 		status = http.StatusBadRequest
 	case ErrDischargeRequired, ErrInteractionRequired:
-		status = http.StatusProxyAuthRequired
+		switch errorBody.version {
```

Comment:
I might be wrong but I can't see where errorBody.version is set. won't this always be 0?